web/src/settings_users.js

```diff
@@ -136,13 +136,19 @@ function role_selected_handler(event, dropdown, widget) {
 }
 
 function get_roles() {
-    return [
-        {unique_id: "0", name: $t({defaultMessage: "All roles"})},
-        ...Object.values(settings_config.user_role_values).map((user_role_value) => ({
-            unique_id: user_role_value.code.toString(),
-            name: user_role_value.description,
-        })),
-    ];
+    const roles = [{unique_id: "0", name: $t({defaultMessage: "All roles"})}];
+    const user_role_array = Object.values(settings_config.user_role_values);
+
+    // We traverse the array in reverse, since we want to rank the roles by
+    // their privileges, e.g. All roles, Owner, Administrator, etc. We do this
+    // because one of the major use cases of the role filter is to find admins.
+    for (let index = user_role_array.length - 1; index >= 0; index -= 1) {
+        roles.push({
+            unique_id: user_role_array[index].code.toString(),
+            name: user_role_array[index].description,
+        });
+    }
+    return roles;
```

Comment:
Would it be better to just use https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/reverse before returning, and stick with the map/values approah?

Reply:
@timabbott Do you mean something like 

```
 return [
        {unique_id: "0", name: $t({defaultMessage: "All roles"})},
        ...Object.values(settings_config.user_role_values).map((user_role_value) => ({
            unique_id: user_role_value.code.toString(),
            name: user_role_value.description,
        })).reverse(),
    ];
 ```
 
 Wouldn't it still traverse through the array twice as mentioned in the commit message? Once, during `Array.map()` and once during `Array.reverse()`. Or since it's a small array, we should favour readability here?

Reply:
Yeah it's a 6-element array, so readability should be the priority.

Reply:
https://zulip.readthedocs.io/en/latest/subsystems/performance.html talks a little about this, but micro-optimizations that cost readability are usually not worth it; and this one is probably measured in nanoseconds for a thing that happens once per user who opens this panel.

Reply:
Makes sense, I've updated the PR!